main.go

```diff
@@ -18,11 +18,24 @@ package main
 
 import (
 	"fmt"
+	"io"
 	"log"
 	"os"
 )
 
+var (
+	fs     fileSystem
+	runner cmdRunner
+	rc     clock
+	out    io.Writer
```

Comment:
I tend to come from the background where I'd rather be passing variables into something rather than requiring globals; but I also understand that golang is not a functional language. 

In this situation, we have variables being declared in our main function, which imo, should be a clean, clear function on what it is doing every step of the way. So we have two choices, that I see:

1. make the tests not depend on external variables; allow them to accept params that make the tests run. 

2. Comment our code better and explain why we are doing this in the main function as it seems to be a bit out of the context of the actual application. Someone reading this through for the first time might be confused why we have `fs` being declared as a variable in the main file yet it's usage is not noted in the file anywhere.

IMO, our main function should only be concerned with the actual application functionality, tests should be orthogonal (or... parallel? I don't know the correct word to use here). I don't know where that opinion falls on the "idiomatic go" side of things. 

I also have a question from your other comment:

> It allows us to mock out our filesystem calls in tests but use the real implementation when running the application.

Why do we need to mock our file system calls? Could we just use real file system calls and rely on real data (was that what we were doing before?) Does mocking the file system provide more reliable testing? 

---

Ok so that was a lot - take the questions with a grain of salt as more so being directed at me trying to learn Golang, and not questioning your methodologies (although I understand if it comes across that way).

I'm onboard with using convey for sure, it seems cool. In this context you've done a lot of hard work that we need, and in the time constraints we might as well kick this off. 

What do you think about being able to move the global vars out of main? I haven't had a super close look at the code but let me know if that's feasible?

I'll be offline for the weekend, but will try and merge this when I get back.